src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -320,4 +326,32 @@ private BracketChecker(String field) {
         }
     }
 
+    private static class HashChecker implements Checker {
```

Comment:
I would refer to # as the number sign, because "hash" reminds me too much of the hash code. Maybe use some more functional name like `BibStringChecker`